src/sentry/digests/notifications.py

```diff
@@ -170,14 +175,40 @@ def build_digest(project: Project, records: Sequence[Record]) -> DigestInfo:
     start = records[-1].datetime
     end = records[0].datetime
 
+    rule_ids: set[int] = set()
+    workflow_ids: set[int] = set()
+
+    for record in records:
+        identifier_key = getattr(record.value, "identifier_key", IdentifierKey.RULE)
+        # record.value is Notification, record.value.rules is Sequence[int]
+        ids_to_add = record.value.rules
+        if identifier_key == IdentifierKey.RULE:
+            rule_ids.update(ids_to_add)
+        elif identifier_key == IdentifierKey.WORKFLOW:
+            workflow_ids.update(ids_to_add)
+
     groups = Group.objects.in_bulk(record.value.event.group_id for record in records)
     group_ids = list(groups)
-    rules = Rule.objects.in_bulk(rule_id for record in records for rule_id in record.value.rules)
+    rules = Rule.objects.in_bulk(rule_ids)
 
-    # TODO(iamrajjoshi): This will only work during the dual write period of the rollout!
     if features.has("organizations:workflow-engine-trigger-actions", project.organization):
         for rule in rules.values():
             rule.data["actions"][0]["legacy_rule_id"] = rule.id
+    # We are only proccessing the workflows in the digest if under the new flag
+    # This should be ok since we should only add workflow_ids to redis when under this flag
+    if features.has("organizations:workflow-engine-ui-links", project.organization):
+        for workflow_id in workflow_ids:
+            workflow = Workflow.objects.get(id=workflow_id)
+            assert (
+                workflow.organization_id == project.organization_id
+            ), "Workflow must belong to Organization"
+            rules[workflow_id] = Rule(
```

Comment:
I'm a little confused by this, isn't `rules` a dict of rule ids to `Rule` objects? Why would the workflow ids correspond to a rule id? Or is it that you're adding rules with a workflow id to that dict?